app/src/main/kotlin/com/mapbox/vision/teaser/utils/PermissionsUtils.kt

```diff
@@ -0,0 +1,49 @@
+package com.mapbox.vision.teaser.utils
+
+import android.app.Activity
+import android.content.pm.PackageManager
+import android.os.Build
+import androidx.core.content.ContextCompat
+import com.mapbox.vision.teaser.MainActivity
+
+private const val PERMISSION_FOREGROUND_SERVICE = "android.permission.FOREGROUND_SERVICE"
+private const val PERMISSIONS_REQUEST_CODE = 123
+
+fun MainActivity.requestPermissionsIfNotGranted(): Boolean {
```

Comment:
Why Main? Perhaps extend `Activity`?

Reply:
> @yuryybk - could we also change sort order in this PR which we discussed this morning?
Fixed!

Reply:
For now It is quite specific simplified code which should be used on MainActivity at the beginning

Reply:
But talking strictly we could call it from every activity and it still will work.
Up to you however.

Reply:
It should split permissions to be usable for other activities. Let's keep it strictly linked to Main for now

Reply:
@yuryybk WDYT about adding those permissions as methods for some utils object instead, e.g.
```
object PermissionUtils { 
  fun requestPermissions(context: Activity) // or MainActivity if we'd like to restrict them anyway

  fun allPermissionsGrantedByRequest

...
}
```

Reply:
Then use of this code will be readable and won't confuse external users, who might read this code,  
e.g. `PermissionUtils.blabla` instead of `blabla`

Reply:
I've used extension to make it linked to the MainActivity.
PermissionUtils more common way and at first I've implemented with PermissionUtils (https://github.com/mapbox/mapbox-vision-android-teaser/pull/205/commits/ba9f0d1116f342738833ef6a804627c07d5c2e1a)
Let's use PermisssionUtils as it is more clear

Reply:
Suppose you can pass `MainActivity` to restrict API with PermisionUtils as well

Reply:
fixed